src/Features/Blockcore.Features.Miner/Api/Controllers/StakingController.cs

```diff
@@ -79,6 +79,31 @@ public IActionResult GetStakingInfo()
             }
         }
 
+        /// <summary>
+        /// Get staking info from the miner.
+        /// </summary>
+        /// <returns>All staking info details as per the GetStakingInfoModel.</returns>
+        [Route("getnetworkstakinginfo")]
+        [HttpGet]
+        public IActionResult GetNetworkStakingInfo()
+        {
+            try
+            {
+                if (!this.fullNode.Network.Consensus.IsProofOfStake)
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.MethodNotAllowed, "Method not allowed", "Method not available for Proof of Stake");
```

Comment:
The error message says PoS wrongly, it should be Proof-of-Work, right?

Reply:
ah well spotted